src/systems/user_commands/UserCommands.cc

```diff
@@ -698,6 +747,44 @@ bool PoseCommand::Execute()
   return true;
 }
 
+//////////////////////////////////////////////////
+PhysicsCommand::PhysicsCommand(msgs::Physics *_msg,
+    std::shared_ptr<UserCommandsInterface> &_iface)
+    : UserCommandBase(_msg, _iface)
+{
+}
+
+//////////////////////////////////////////////////
+bool PhysicsCommand::Execute()
+{
+  auto physicsMsg = dynamic_cast<const msgs::Physics *>(this->msg);
+  if (nullptr == physicsMsg)
+  {
+    ignerr << "Internal error, null physics message" << std::endl;
+    return false;
+  }
+
+  auto worldEntity = this->iface->ecm->EntityByComponents(components::World());
+  if (worldEntity == kNullEntity)
+  {
+    ignmsg << "Failed to find world entity" << std::endl;
+    return false;
+  }
+
+  if (!this->iface->ecm->EntityHasComponentType(worldEntity,
+    components::PhysicsCmd().TypeId()))
+  {
+    auto physics = convert<sdf::Physics>(*physicsMsg);
+    this->iface->ecm->CreateComponent(worldEntity,
+        components::PhysicsCmd(physics));
+    // HACK, the component is meant to be updated in SimulationRunner
+    // auto physicsComponent =
+    //  this->iface->ecm->Component<components::Physics>(worldEntity);
+    // physicsComponent->Data() = components::Physics(physics).Data();
```

Comment:
I guess the last part I'm not sure on my side is if we are OK with the glitch in the main post (when the step size is updated without pressing enter) or if we should add this hack to fix it but at the expense of creating possible glitches if the `PhysicsCmd` component is not processed in `SimulationRunner`, since with this hack the `Physics` component would still be updated and have incorrect information.

Reply:
Fishy. I'd expect the hack not to be necessary. I'd expect `sendPhysics` to be called when the mouse moves away, just as if you pressed enter.

Reply:
This was fixed when moving the updating of the physics parameters to before the systems are updated. I'm guessing that the GUI system was sent the old parameters in its `Update()` call, then immediately after updated to the new one in the `UpdatePhysicsParams()` call, resulting in this glitch.